### PR TITLE
Remove Order constraints from SortedMap and NonEmptyMap instances

### PR DESCRIPTION
**Pull request by travisbrown**
These are @joroKr21's commits from #3397, but rebased to resolve a conflict. (I'm opening this as a new PR since I messed up the conflict resolution in #3397, but the commits / PR credit will go to @joroKr21.)

### CODE REVIEW

File: core/src/main/scala/cats/data/NonEmptyMapImpl.scala [hunk 1]
```diff
@@ -14,13 +13,21 @@ private[data] object NonEmptyMapImpl extends NonEmptyMapInstances with Newtype2
   private[data] def unwrap[K, A](m: Type[K, A]): SortedMap[K, A] =
     m.asInstanceOf[SortedMap[K, A]]
 
-  def fromMap[K: Order, A](as: SortedMap[K, A]): Option[NonEmptyMap[K, A]] =
+  def fromMap[K, A](as: SortedMap[K, A]): Option[NonEmptyMap[K, A]] =
     if (as.nonEmpty) Option(create(as)) else None
 
-  def fromMapUnsafe[K: Order, A](m: SortedMap[K, A]): NonEmptyMap[K, A] =
+  @deprecated("Use fromMap override without Order", "2.2.0-M2")
```
Comment (rossabaker) [inline core/src/main/scala/cats/data/NonEmptyMapImpl.scala:19]: Shouldn't this (and others) read 2.2.0-M3?

Ah, the joys of long-running PRs.

Comment (travisbrown) [reply core/src/main/scala/cats/data/NonEmptyMapImpl.scala:19]: Right, @barambani noted that in #3397 but I didn't want to change @joroKr21's commits more than the merge required. I could go ahead and fix that here, though.

Comment (LukaJCB) [reply core/src/main/scala/cats/data/NonEmptyMapImpl.scala:19]: easy fix, let's do it

File: core/src/main/scala/cats/instances/sortedMap.scala [hunk 2]
```diff
@@ -19,23 +17,22 @@ trait SortedMapInstances extends SortedMapInstances2 {
...
-        m.iterator
-          .map { case (a, b) => showA.show(a) + " -> " + showB.show(b) }
-          .mkString("SortedMap(", ", ", ")")
-    }
+  implicit def catsStdShowForSortedMap[A, B](implicit showA: Show[A], showB: Show[B]): Show[SortedMap[A, B]] =
+    _.iterator
+      .map { case (a, b) => showA.show(a) + " -> " + showB.show(b) }
+      .mkString("SortedMap(", ", ", ")")
+
+  @deprecated("Use catsStdShowForSortedMap override without Order", "2.2.0-M2")
```
Comment (barambani) [inline core/src/main/scala/cats/instances/sortedMap.scala:25]: there are still a few. They can be a follow up though, as long as it happens before M3

**Comment by travisbrown** [comment]
Just fixed the rest of the deprecation versions. Will need one more sign-off since that invalidated the reviews, then I'll merge (with a merge commit to make sure @joroKr21 is credited) when it's green.

**Comment by travisbrown** [comment]
@barambani @LukaJCB Mind a quick re-approval?

**Comment by joroKr21** [comment]
Thanks for picking this up 👍

**Comment by johnynek** [comment]
How can this be binary compatible? The implicits lost a parameter right?

Or is this for a binary breaking release?

**Comment by joroKr21** [comment]
> How can this be binary compatible? The implicits lost a parameter right?

The implicits that lost a parameter (instead of being deprecated like others) are not yet released in a stable version.

**Pull request merged**